### ISSUE TITLE
[JENKINS-47667] "Option" to disable every publishers/reporters

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/pipeline-maven-plugin#1087]
<p>Hello,</p>

<p>In our main build, we build in parallel with many concerns in mind: one branch (of the parallel builds) is the main build where we leave everything enabled, and we have dedicated branches for coverage, release checking, etc.</p>

<p>As we already do the reporting on the main branch, we disable many reporters on the non-main branches. Currently, it's a bit cumbersome to do this as we have to add as many reporters as there are. Also, if new ones get added in the future, we'll have to add them too to avoid any spurious reporting to starting occurring.</p>

<p>So, it would be nice to have an option to reverse the default logic: include no reporter at all by default, and then let me specify the ones I need.</p>

<p>Something like:</p>

```
withMaven(...',
  options: ['disableAll: true', openTasksPublisher(disabled: false)]) {
    sh 'mvn ...'
}

```
<p>To get only the Tasks publisher report to be created.</p>

<p>Thanks!</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/batmat.png" title="batmat's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=batmat">batmat</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-47667" target="_blank">"Option" to disable every publishers/reporters</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/alobato.png" title="alobato's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=alobato">alobato</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Minor
<li><b>component(s)</b>: pipeline-maven-plugin
<li><b>resolution</b>: Duplicate
<li><b>resolved</b>: 2017-11-02T13:16:52+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 1
<li><b>imported</b>: 20251211-223321
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Hello,</p>

<p>In our main build, we build in parallel with many concerns in mind: one branch (of the parallel builds) is the main build where we leave everything enabled, and we have dedicated branches for coverage, release checking, etc.</p>

<p>As we already do the reporting on the main branch, we disable many reporters on the non-main branches. Currently, it's a bit cumbersome to do this as we have to add as many reporters as there are. Also, if new ones get added in the future, we'll have to add them too to avoid any spurious reporting to starting occurring.</p>

<p>So, it would be nice to have an option to reverse the default logic: include no reporter at all by default, and then let me specify the ones I need.</p>

<p>Something like:</p>
<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>withMaven(...',
          options: ['disableAll: true', openTasksPublisher(disabled: false)]) {
    sh 'mvn ...'
}
</pre>
</div></div>
<p>To get only the Tasks publisher report to be created.</p>

<p>Thanks!</p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `duplicates` from Jira: <a class="jira-relationship" href="https://issue-redirect.jenkins.io/issue/46785">JENKINS-46785</a>]</i>
